_source/_posts/2023-03-20-spa-auth-tokens.md

```diff
@@ -0,0 +1,228 @@
+
+While each step of this OAuth flow to get the tokens is critical to ensure a secure authentication and authorization process, let's inspect the two requests in more detail. 
+
+## Create an Authorization server and view the well-known endpoints 
+
+We'll start by setting up an Authorization server in Okta and use the [OpenID Connect Debugger](https://oidcdebugger.com/) tool to inspect the Network requests.
+
+{% include setup/cli.md type="spa" loginRedirectUri="https://oidcdebugger.com/debug" %}
+
+Make note of the `Issuer` and `Client ID`. In the upcoming steps, you'll need those values to configure Okta in your SPA client.
+
+Authorization servers have standard, public endpoints for discovery by clients. In our case, because we are using OAuth 2.0 + OIDC, we have two discovery endpoints.
+
+### OAuth Authorization Server Discovery
+
+The OAuth specification requires there is a standardized mechanism for clients to discover metadata about the Authorization Server at the `{Issuer}/.well-known/oauth-authorization-server`. Replace the `{Issuer}` with the value from Okta CLI and navigate to the path in your browser.
+
+You should see JSON formatted data with information that provides the necessary information to interact with the Authorization server:
+
+{% img blog/spa-auth-tokens/well-known-oauth.jpg alt:"Example OAuth well-known discovery document" width:"800" %}{: .center-image }
+
+The OAuth discovery document doesn't have any user information. It only has information about the resources the user has access to.
+
+We have the endpoints for the authorization and token requests, which we'll inspect momentarily.
+
+### OIDC Discovery
+
+The OpenID Connect specification also requires a standardized mechanism for client discovery. You can find it at `{Issuer}/.well-known/openid-configuration`. Open up a browser tab to the OIDC discovery endpoint. It is also JSON formatted data that looks something like this:
+
+{% img blog/spa-auth-tokens/well-known-oauth.jpg alt:"Example OIDC well-known discovery document" width:"800" %}{: .center-image }
+
+Like the OAuth discovery response, we have the endpoints for the authorization and token requests. We also have a user info endpoint to query for user information. If you find the property for `claims_supported`, you'll see the claims cover various identifying information about the user. 
+
+## Debug the Authorization Code and Token requests 
```

Comment:
```suggestion
## Debug the Authorization Code and Token Requests 
```

Reply:
I think this one doesn't need capitalization. As a matter of fact, I'll change this whole thing to 
> Debug the authorization code and token requests